### PR TITLE
Allow adding children to DynamicComponent

### PR DESCRIPTION
**Pull request by Taig**
That was a bit of guesswork for me, you should therefore check it very carefully.

### CODE REVIEW

File: src/main/scala/com/github/ahnfelt/react4s/Component.scala [hunk 1]
```diff
@@ -90,7 +90,7 @@ sealed trait ElementOrComponent extends Tag {
 }
 
 /** Wraps a React component written in JavaScript. Example: ```DynamicComponent(js.Dynamic.global.MyJsComponent, js.Dictionary("label" -> "Go!"))``` */
-final case class DynamicComponent(componentClass : Any, props : Any, key : Option[String] = None, ref : Option[Any => Unit] = None) extends ElementOrComponent {
+final case class DynamicComponent(componentClass : Any, props : Any, children : Seq[Tag], key : Option[String] = None, ref : Option[Any => Unit] = None) extends ElementOrComponent {
```
Comment (Taig) [inline src/main/scala/com/github/ahnfelt/react4s/Component.scala]: I'd kinda like to have empty defaults á la `props : js.Dictionary = js.Dicitionary.empy` and `children : Seq[Tag] = Seq.empty` here, but didn't dare to pull in `js.*`.

**Comment by Ahnfelt** [comment]
Looks good to me! A minor thing, I think `dynamic.children.isDefined` needs to be checked as well here:
https://github.com/Taig/react4s/blob/7cf77be4df9017c723e84b8d8432753af4c601f6/src/main/scala/com/github/ahnfelt/react4s/ReactBridge.scala#L100

File: src/main/scala/com/github/ahnfelt/react4s/Component.scala (hunk 1 above)
Comment (Ahnfelt) [inline src/main/scala/com/github/ahnfelt/react4s/Component.scala]: Hmm, couldn't you have `children : Seq[Tag] = Seq.empty` here without pulling in `js.*`?

Comment (Taig) [inline src/main/scala/com/github/ahnfelt/react4s/Component.scala]: For the children, yes. For the props, no. So I preferred to stay consistent with what you had so far.

Comment (Ahnfelt) [inline src/main/scala/com/github/ahnfelt/react4s/Component.scala]: Ah, I see. But `props` may actually contain the `children`, `key` and `ref` fields already - that's why `key` and `ref` have default values, and I think it makes sense to give `children : Seq[Tag] = Seq.empty` in the same vein.

**Comment by Ahnfelt** [comment]
Great, thank you :)

Formalia (ugh): Do you have the right to & agree to release the code contained in this pull request under the MIT License: https://opensource.org/licenses/MIT?

**Comment by Taig** [comment]
Yes

**Comment by Ahnfelt** [comment]
Thank you, I've released a new version with the latest changes: 0.4-SNAPSHOT

**Pull request merged**